x-pack/plugins/cases/public/methods/can_use_cases.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ApplicationStart } from 'kibana/public';
+import { CasePermissionOptions } from '../types';
+
+export function canUseCases(capabilities: ApplicationStart['capabilities']) {
+  const defaultCaseOptions = {
+    owners: [],
+    crudLevel: 'read',
+  };
+  return (options: CasePermissionOptions): boolean => {
+    const casePermissions = { ...defaultCaseOptions, ...options };
+    casePermissions.owners = casePermissions.owners.length
+      ? casePermissions.owners
+      : ['securitySolutionCases', 'observabilityCases'];
+
+    for (const type of casePermissions.owners) {
+      if (
+        Object.hasOwnProperty.call(capabilities, type) &&
```

Comment:
Just curious is there an advantage to this style vs `capabilities.hasOwnProperty(type)` ? does the first protect against `capabilities` being `null/undefined`?

Reply:
I had seen @cnasikas using this style so I opted to as well. I wasnt sure why though. What was the reason behind it?

Reply:
`Object.hasOwnProperty.call` is safer from `myObject.hasOwnProperty` because someone can do

```
var myObject = {
    hasOwnProperty:() => {
        return true;
    }
};
```

and override the `hasOwnProperty` function. By doing `Object.hasOwnProperty.call` you really on the `Object` and not to your instance.

Also if you create an object with `const myObject = Object.create(null)` then `myObject.hasOwnProperty` will be `undefined`.